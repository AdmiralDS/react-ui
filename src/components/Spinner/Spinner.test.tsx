import React from 'react';
import { render } from '@testing-library/react';

import { Spinner } from './';

describe('Spinner', () => {
  it('should render default component', () => {
    const wrapper = render(<Spinner />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with small dimension', () => {
    const wrapper = render(<Spinner dimension="s" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with inverse color', () => {
    const wrapper = render(<Spinner inverse />);
    expect(wrapper).toMatchSnapshot();
  });
});
