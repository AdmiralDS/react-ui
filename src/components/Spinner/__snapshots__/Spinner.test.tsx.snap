// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render component with inverse color 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  position: relative;
  width: 24px;
  height: 24px;
}

.c0 svg {
  -webkit-animation: cilQsd 1s linear infinite;
  animation: cilQsd 1s linear infinite;
}

.c0 svg path {
  fill: #FFFFFF;
}

<body>
    <div>
      <div
        aria-live="assertive"
        class="c0"
        role="alert"
      >
        <div />
      </div>
    </div>
  </body>,
  "container": .c0 {
  position: relative;
  width: 24px;
  height: 24px;
}

.c0 svg {
  -webkit-animation: cilQsd 1s linear infinite;
  animation: cilQsd 1s linear infinite;
}

.c0 svg path {
  fill: #FFFFFF;
}

<div>
    <div
      aria-live="assertive"
      class="c0"
      role="alert"
    >
      <div />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Spinner should render component with small dimension 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  position: relative;
  width: 16px;
  height: 16px;
}

.c0 svg {
  -webkit-animation: cilQsd 1s linear infinite;
  animation: cilQsd 1s linear infinite;
}

.c0 svg path {
  fill: #0D69F2;
}

<body>
    <div>
      <div
        aria-live="assertive"
        class="c0"
        role="alert"
      >
        <div />
      </div>
    </div>
  </body>,
  "container": .c0 {
  position: relative;
  width: 16px;
  height: 16px;
}

.c0 svg {
  -webkit-animation: cilQsd 1s linear infinite;
  animation: cilQsd 1s linear infinite;
}

.c0 svg path {
  fill: #0D69F2;
}

<div>
    <div
      aria-live="assertive"
      class="c0"
      role="alert"
    >
      <div />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Spinner should render default component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  position: relative;
  width: 24px;
  height: 24px;
}

.c0 svg {
  -webkit-animation: cilQsd 1s linear infinite;
  animation: cilQsd 1s linear infinite;
}

.c0 svg path {
  fill: #0D69F2;
}

<body>
    <div>
      <div
        aria-live="assertive"
        class="c0"
        role="alert"
      >
        <div />
      </div>
    </div>
  </body>,
  "container": .c0 {
  position: relative;
  width: 24px;
  height: 24px;
}

.c0 svg {
  -webkit-animation: cilQsd 1s linear infinite;
  animation: cilQsd 1s linear infinite;
}

.c0 svg path {
  fill: #0D69F2;
}

<div>
    <div
      aria-live="assertive"
      class="c0"
      role="alert"
    >
      <div />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
